pkg/auth/providers/github/github_client.go

```diff
@@ -118,11 +114,54 @@ func (g *GClient) getTeams(githubAccessToken string, config *v32.GithubConfig) (
 			logrus.Errorf("Github getGithubTeams: received error unmarshalling teams array, err: %v", err)
 			return teams, err
 		}
-		for _, teamObj := range teamObjs {
-			teams = append(teams, teamObj)
+		teams = append(teams, teamObjs...)
+
+	}
+	return teams, nil
+}
+
+// getOrgTeams returns the teams belonging to an organization.
+func (g *GClient) getOrgTeams(githubAccessToken string, config *v32.GithubConfig, org Account) ([]Account, error) {
+	url := fmt.Sprintf(g.getURL("ORG_TEAMS", config), org.Login)
+	responses, err := g.paginateGithub(githubAccessToken, url)
+	if err != nil {
+		logrus.Errorf("Github getGithubTeams: GET url %v received error from github, err: %v", url, err)
+		return nil, err
+	}
+
+	var teams, respTeams []Account
```

Comment:
This could be quite expensive in terms of API calls, you have an unknown number of pages to fetch.

Also, you have to load them _all_ to filter on the client side, even if the only matching one is on the first page, which feels wasteful?

Maybe this this could be better handled with some GraphQL?

```graphql
query {
  organization(login: "rancher") {
    teams(query: "ci", first: 10) {
      edges {
        node {
          name
        }
      }
    }
  }
}
```

You'd still need to get pages, but each page would be relevant? https://docs.github.com/en/graphql/guides/using-pagination-in-the-graphql-api

Reply:
I don't think graphql is any faster, most likely the opposite (at least that was my observation). And yes you still need to paginate regardless.
But most importantly the intent of this particular PR is not to redesign how this auth provider fetches the data from the GH api but rather implement a missing functionality and do so using existing pieces.